### DISCUSSION TITLE
How to add vulnerability for internal component

### DISCUSSION THREAD

**Discussion by iamrahul127** [DependencyTrack/dependency-track#2159] [Q&A] [answered]
Hi, I have gone through https://github.com/DependencyTrack/dependency-track/issues/96 and can see that, starting v 4.5.0 we have ability to add internal vulnerability. Problem is, even after giving "VULNERABILITY_MANAGEMENT" permission, user is not getting 'Create Vulnerability' button on the Vulnerabilities screen. Screen Shots as below.

1. Administrator has "VULNERABILITY_MANAGEMENT" permission.
![image](https://user-images.githubusercontent.com/24877322/201886983-499b63b4-709c-48a7-a7e8-26cca709ca5d.png)
2. User is part of Administrator team.
![image](https://user-images.githubusercontent.com/24877322/201887506-b8434a42-1726-42ba-ab26-dbb8fb404ad7.png)
3. No "Create" button on Vulnerabilities screen. 
![image](https://user-images.githubusercontent.com/24877322/201888191-f7088dd6-ed67-485e-8a2b-c70975e4916c.png)

**Comment by syalioune** [comment]
Hello @iamrahul127 
It can be a session issue, the token generated by DT upon connection has a lifespan of 7d and contains the user permissions.
Can you try logout and login again ?

**Reply by iamrahul127** [reply]
Hi @syalioune, Yes I am aware of long token validity. I have tried log off and login several times.

Is there anything else I should be trying?

**Reply by syalioune** [reply]
That's odd. Can you check the permissions in the token after login ?
Just in case, you can retrieve it in your browser session storage and decode it using https://jwt.io/

**Reply by iamrahul127** [reply]
I see the permission included in token. Now it more strange. 

{
  "sub": "Rahul.Mahulkar",
  "iss": "Dependency-Track",
  "iat": 1668589499,
  "exp": 1669194299,
  "permissions": "ACCESS_MANAGEMENT,BOM_UPLOAD,POLICY_MANAGEMENT,POLICY_VIOLATION_ANALYSIS,PORTFOLIO_MANAGEMENT,PROJECT_CREATION_UPLOAD,SYSTEM_CONFIGURATION,VIEW_POLICY_VIOLATION,VIEW_PORTFOLIO,VIEW_VULNERABILITY,VULNERABILITY_ANALYSIS,VULNERABILITY_MANAGEMENT",
  "idp": "OPENID_CONNECT"
}

**Comment by syalioune** [comment]
Maybe some kind of browser cache 😅 ?

**Reply by iamrahul127** [reply]
Tried in incognito mode  and in chrome/Firefox but still not seeing the button. :) Is it a bug then?

**Reply by syalioune** [reply]
Either a bug or your frontend version is not correct.
@nscuro what do you think ?

**Reply by nscuro** [reply]
Checking the frontend version would by my next suggestion, yes. @iamrahul127, which version of DT are you using, and how did you deploy it?

**Comment by iamrahul127** [comment, marked as answer]
@nscuro I am using version 4.5.0. Screenshot as below.
![image](https://user-images.githubusercontent.com/24877322/202236853-eddfa26d-baac-4fa1-bb95-6cab5d0bc007.png)

I have installed it using helm chart.  Chart details as below.

![image](https://user-images.githubusercontent.com/24877322/202237146-69b776b4-ab03-4b3e-8933-0dfb89ff92dd.png)

**Reply by nscuro** [reply]
Can you please click on the `Dependency-Track` link from your first screenshot? It will display the concrete versions of both API server and frontend. Version 1.4.0 of the chart you're using wrongly shipped frontend version 4.**4**.0 with API server 4.5.0, see https://github.com/evryfs/helm-charts/blob/dependency-track-1.4.0/charts/dependency-track/values.yaml#LL18C12-L18C12

**Reply by iamrahul127** [reply]
You are bang on target. Here you go.
![image](https://user-images.githubusercontent.com/24877322/202240092-f869a7b7-4ef5-4a3b-983b-38a75baae5f5.png)

I hope upgrading to latest version 1.5.4 will fix this issue.

**Reply by iamrahul127** [reply]
Upgrade to latest worked.

![image](https://user-images.githubusercontent.com/24877322/202253098-ab5f0ff7-7984-423d-b0cb-2ccaf1df1ffa.png)
